File: Lab_4/Views/app/src/main/java/com/example/lab_4/MainActivity.java
package com.example.lab_4;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnListView = findViewById(R.id.btnListView);
        Button btnGridView = findViewById(R.id.btnGridView);
        Button btnSpinner = findViewById(R.id.btnSpinner);
        Button btnAutoComplete = findViewById(R.id.btnAutoComplete);
        btnListView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, ListViewActivity.class);
                startActivity(intent);
            }
        });
        btnAutoComplete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, AutoCompleteActivity.class);
                startActivity(intent);
            }
        });
        btnGridView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, GridViewActivity.class);
                startActivity(intent);
            }
        });

        btnSpinner.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, SpinnerActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: Lab_4/Donation_Layouts/app/src/main/java/com/example/donationlayout/MainActivity.kt
package com.example.donationlayout

import android.os.Bundle
import android.view.View
import android.widget.ProgressBar
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.donationlayout.ui.theme.DonationLayoutTheme
import com.example.donationlayout.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private val TAG = "ActivityMain"
    private lateinit var binding: ActivityMainBinding;
    private var totalAmount = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        val numberPicker = binding.numberPicker
        val txtUpdate = binding.exitTextAmount
        val progressBar = findViewById<ProgressBar>(R.id.progressBar1)
        val btnDonate = binding.btnDonate
        val txtTotal = binding.textTotal
        val addAmount = binding.exitTextAmount

        progressBar.visibility = View.VISIBLE

        numberPicker.minValue=1
        numberPicker.maxValue=100

        numberPicker.setOnValueChangedListener { _, _, _ ->
            progressBar.progress = numberPicker.value
        }
        btnDonate.setOnClickListener {
            totalAmount += Integer.parseInt(addAmount.text.toString())
            txtTotal.text = "Total amount is $totalAmount"
        }
    }
}
